task_interface_mapreduce.go

```diff
@@ -0,0 +1,15 @@
+package taskgraph
+
+// MapreduceTask is a logic repersentation of a computing unit in mapreduce framework.
+// Each task contain at least one Node.
+
+type MapreduceTask interface {
```

Comment:
Ok. I think I get Emit() and Collect() here.
Those are implementation details that might vary.
Let's first define the user interface of MapReduce Framework.